app/src/main/java/com/babylon/wallet/android/data/repository/cache/database/Joins.kt

```diff
@@ -0,0 +1,153 @@
+package com.babylon.wallet.android.data.repository.cache.database
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Index
+import com.babylon.wallet.android.data.gateway.extensions.amount
+import com.babylon.wallet.android.data.gateway.extensions.amountDecimal
+import com.babylon.wallet.android.data.gateway.extensions.vaultAddress
+import com.babylon.wallet.android.data.gateway.generated.models.FungibleResourcesCollectionItem
+import com.babylon.wallet.android.data.gateway.generated.models.NonFungibleResourcesCollectionItem
+import com.babylon.wallet.android.data.gateway.generated.models.StateNonFungibleDetailsResponseItem
+import com.babylon.wallet.android.data.repository.cache.database.NFTEntity.Companion.asEntity
+import com.babylon.wallet.android.data.repository.cache.database.ResourceEntity.Companion.asEntity
+import java.math.BigDecimal
+
+@Entity(
+    primaryKeys = ["account_address", "resource_address"],
+    foreignKeys = [
+        ForeignKey(
+            entity = AccountEntity::class,
+            parentColumns = ["address"],
+            childColumns = ["account_address"],
+            onDelete = ForeignKey.CASCADE,
+        ),
+        ForeignKey(
+            entity = ResourceEntity::class,
+            parentColumns = ["address"],
+            childColumns = ["resource_address"]
+        )
+    ]
+)
+data class AccountResourceJoin(
```

Comment:
So just to understand it properly what is the purpose of those Joins data classes ? Are they supposed to be db models ?

Reply:
Yes `*Entity` are models of actual resources, components & accounts. `*Join` is the modeling of their relationship. Currently in code a `FungibleResource` hides this relationship with the `ownedAmount` being part of the resource, but you have to remember that this resource belongs to an account in AccountWithAssets. 

Having a simple resource saved in database, needs a join to describe the relationship of the resource with an account.

Reply:
Perfect makes sense thanks 👍